src/main/kotlin/com/chattriggers/ctjs/minecraft/wrappers/entity/Entity.kt

```diff
@@ -296,7 +300,57 @@ open class Entity(val entity: MCEntity) {
         getWorld().getChunkFromChunkCoords(entity.chunkCoordX, entity.chunkCoordZ)
     )
 
+    /**
+     * @return [RayTraceResult] of the block that is being looking at
+     * within the specified distance.
+     */
+    @JvmOverloads
+    fun rayTraceBlock(maxDistance: Double = 200.0, partialTicks: Float = 1f) = RayTraceResult(entity.rayTrace(maxDistance, partialTicks))
+
+    /**
+     * @return [RayTraceResult] of the entity that is being looking at
+     * within the specified distance.
+     */
+    @JvmOverloads
+    fun rayTraceEntity(maxDistance: Double = 200.0, partialTicks: Float = 1f): RayTraceResult {
+        val eyePosition = entity.getPositionEyes(partialTicks)
+        val lookDirection = entity.getLook(partialTicks)
+        val lookingAt = eyePosition.addVector(
+            lookDirection.xCoord * maxDistance,
+            lookDirection.yCoord * maxDistance,
+            lookDirection.zCoord * maxDistance
+        )
+
+        val blockHit = rayTraceBlock(maxDistance, partialTicks)
+        var entityHit: MCEntity? = null
+        var hitVec: Vec3 = eyePosition
+
+        getWorld().getEntitiesWithinAABBExcludingEntity(
+            entity,
+            entity.entityBoundingBox.addCoord(
+                lookDirection.xCoord * maxDistance,
+                lookDirection.yCoord * maxDistance,
+                lookDirection.zCoord * maxDistance
+            )
+        ).forEach { entityFound ->
+            entityFound.entityBoundingBox.calculateIntercept(eyePosition, lookingAt)?.also { result ->
```

Comment:
Maybe I'm missing it, but is this guaranteed to return the closest entity? Like if you're looking through multiple, it seems like it'll just pick the one that happened to be returned last from `getEntitiesWithinAABBExcludingEntity`